book/src/input/how_to_define_custom_control_bindings.md

```diff
@@ -0,0 +1,172 @@
+# How to Define Custom Control Bindings
+
+Instead of using `StringBindings` for an `InputBundle` you probably want to use a custom type in production, as `StringBindings` are mainly meant to be used for prototyping and not very efficient.
+
+Using a custom type to handle input instead of using `String` has many advantages:
+
+* A `String` uses quite a lot of memory compared to something like an enum.
+* Inputting a `String` when retrieving input data is error-prone if you mistype it or change the name.
+* You will be able to add additional information to a custom type.
+
+## Defining a Custom Type
+
+Defining a custom type for the `InputBundle` is done by implementing the `BindingTypes` trait. This trait contains two types, an `Axis` type and an `Action` type. These types are usually defined as enums, but you can use whatever you want.
+
+```rust,edition2018,no_run,noplaypen
+use serde::{Serialize, Deserialize};
+use amethyst::input::{BindingTypes, Bindings};
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+enum AxisBinding {
+  Horizontal,
+  Vertical,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+enum ActionBinding {
+  Shoot,
+}
+
+impl Display for AxisBinding {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Display for ActionBinding {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+struct MovementBindingTypes;
+
+impl BindingTypes for MovementBindingTypes {
+  type Axis = AxisBinding;
+  type Action = ActionBinding;
+}
+```
+
+The `Axis` and `Action` type both need to derive all the traits listed above, the first five are used by Amethyst and the last two are for reading and writing to files correctly. They also need to implement `Display` if you want to add a bindings config file.
+
+For serializing and deserializing you need to add [serde](https://crates.io/crates/serde) to the dependencies like this (note that there might be a newer version available, that is also compatible):
+
+```rust,ignore
```

Comment:
Should be `toml` instead

Reply:
I took it from the getting started, so we should change it to `toml` there too then.

Reply:
oh yes, thanks!